hamocc/aufr_bgc.F90

```diff
@@ -466,175 +460,158 @@ SUBROUTINE AUFR_BGC(kpie,kpje,kpke,ntr,ntrbgc,itrbgc,trc,               &
+            locetra(:,:,kpke+1:2*kpke,isco214)=ocetra(:,:,:,isco214)
+            ! Initialise 13C and 14C fields in the same way as in beleg.F90
+            DO k=1,2*kpke
+            DO j=1,kpje
+            DO i=1,kpie
+               IF(omask(i,j) .GT. 0.5) THEN
+                  ! 13C is read in as delta13C, convert to 13C using model restart total C
+                  beta13=locetra(i,j,k,isco213)/1000.+1.
+                  locetra(i,j,k,isco213)=locetra(i,j,k,isco212)*beta13*re1312/(1.+beta13*re1312)
+                  
+                  ! 14C is read in as delta14C, convert to 14C using model restart total C, 
+                  ! normalize 14C by c14fac to prevent numerical errors
+                  beta14=locetra(i,j,k,isco214)/1000.+1.
+                  locetra(i,j,k,isco214)=locetra(i,j,k,isco212)*beta14*re14to/c14fac
+                  
+                  ! Initialise the remaining 13C and 14C fields, using the restart isco212 field
+                  rco213=locetra(i,j,k,isco213)/(locetra(i,j,k,isco212)+safediv)
+                  rco214=locetra(i,j,k,isco214)/(locetra(i,j,k,isco212)+safediv)
+                  locetra(i,j,k,idoc13)=locetra(i,j,k,idoc)*rco213*bifr13
+                  locetra(i,j,k,idoc14)=locetra(i,j,k,idoc)*rco214*bifr14
+                  locetra(i,j,k,iphy13)=locetra(i,j,k,iphy)*rco213*bifr13
+                  locetra(i,j,k,iphy14)=locetra(i,j,k,iphy)*rco214*bifr14
+                  locetra(i,j,k,izoo13)=locetra(i,j,k,izoo)*rco213*bifr13
+                  locetra(i,j,k,izoo14)=locetra(i,j,k,izoo)*rco214*bifr14
+                  locetra(i,j,k,idet13)=locetra(i,j,k,idet)*rco213*bifr13
+                  locetra(i,j,k,idet14)=locetra(i,j,k,idet)*rco214*bifr14
+                  locetra(i,j,k,icalc13)=locetra(i,j,k,icalc)*rco213
+                  locetra(i,j,k,icalc14)=locetra(i,j,k,icalc)*rco214
+               ENDIF
+            ENDDO
+            ENDDO
+            ENDDO
+            
+            if (.not. use_sedbypass) then
+               ! Burial fields for c-isotopes still missing
+               DO  k=1,2*ks
+               DO  j=1,kpje
+               DO  i=1,kpie 
+                  IF(omask(i,j) .GT. 0.5) THEN
+                     rco213=ocetra(i,j,kbo(i,j),isco213)/(ocetra(i,j,kbo(i,j),isco212)+safediv)
+                     rco214=ocetra(i,j,kbo(i,j),isco214)/(ocetra(i,j,kbo(i,j),isco212)+safediv)
+                     powtra2(i,j,k,ipowc13)=powtra2(i,j,k,ipowaic)*rco213
+                     powtra2(i,j,k,ipowc14)=powtra2(i,j,k,ipowaic)*rco214
+                     sedlay2(i,j,k,issso13)=sedlay2(i,j,k,issso12)*rco213*bifr13
+                     sedlay2(i,j,k,issso14)=sedlay2(i,j,k,issso12)*rco214*bifr14
+                     sedlay2(i,j,k,isssc13)=sedlay2(i,j,k,isssc12)*rco213
+                     sedlay2(i,j,k,isssc14)=sedlay2(i,j,k,isssc12)*rco214
+                  ENDIF
+               ENDDO
+               ENDDO
+               ENDDO
```

Comment:
Loop for burial variables should be added here (see comment above)!

Reply:
Done. Thanks for catching this. It was not picked up in my git merge.